src/snowflake/cli/plugins/nativeapp/codegen/snowpark/python_processor.py

```diff
@@ -0,0 +1,276 @@
+    PathMapping,
+    Processor,
+)
+from snowflake.cli.api.utils.rendering import jinja_render_from_file
+from snowflake.cli.plugins.nativeapp.artifacts import (
+    is_glob,
+    resolve_without_follow,
+)
+from snowflake.cli.plugins.nativeapp.codegen.artifact_processor import (
+    ArtifactProcessor,
+    MissingProjectDefinitionPropertyError,
+)
+from snowflake.cli.plugins.nativeapp.codegen.sandbox import (
+    ExecutionEnvironmentType,
+    SandboxExecutionError,
+    execute_script_in_sandbox,
+)
+from snowflake.cli.plugins.nativeapp.codegen.snowpark.extension_function_utils import (
+    _enrich_entity,
+)
+from snowflake.cli.plugins.nativeapp.utils import (
+    filter_files,
+    get_all_file_paths_under_dir,
+    is_python_file,
+)
+
+DEFAULT_TIMEOUT = 30
+
+
+def _determine_virtual_env(processor: Processor) -> Dict[str, Any]:
+    """
+    Determines a virtual environment to run the Snowpark processor in, either through the project definition or by querying the current environment.
+    """
+    if (processor.properties is None) or ("env" not in processor.properties):
+        return {}
+
+    env_props = processor.properties["env"]
+    env_type = env_props.get("type", None)
+
+    if env_type is None:
+        return {}
+
+    if env_type.upper() == ExecutionEnvironmentType.CONDA.name:
+        env_name = env_props.get("name", None)
+        if env_name is None:
+            raise MissingProjectDefinitionPropertyError(
+                "No name found in project definition file for the conda environment to run the Snowpark processor in. Will attempt to auto-detect the current conda environment."
+            )
+        return {"env_type": ExecutionEnvironmentType.CONDA, "name": env_name}
+    elif env_type.upper() == ExecutionEnvironmentType.VENV.name:
+        env_path = env_props.get("path", None)
+        if env_path is None:
+            raise MissingProjectDefinitionPropertyError(
+                "No path found in project definition file for the conda environment to run the Snowpark processor in. Will attempt to auto-detect the current venv path."
+            )
+        return {
+            "env_type": ExecutionEnvironmentType.VENV,
+            "path": env_path,
+        }
+    elif env_type.upper() == ExecutionEnvironmentType.CURRENT.name:
+        return {
+            "env_type": ExecutionEnvironmentType.CURRENT,
+        }
+    return {}
+
+
+def _execute_in_sandbox(
+    py_file: str, deploy_root: Path, kwargs: Dict[str, Any]
+) -> Optional[List[Dict[str, Any]]]:
+    # Create the code snippet to be executed in the sandbox
+    script_source = jinja_render_from_file(
+        template_path=Path("./callback_source.py.jinja"), data={"py_file": py_file}
+    )
+
+    try:
+        completed_process = execute_script_in_sandbox(
+            script_source=script_source,
+            cwd=deploy_root,
+            timeout=DEFAULT_TIMEOUT,
+            **kwargs,
+        )
+    except SandboxExecutionError as sdbx_err:
```

Comment:
If we hit one of these, it's likely to affect all python files the same way. The user will then see the same line for every single script file. Fine for now, but maybe we should change the exceptions a bit so that we can distinguish between "the sandboxing is broken" and "this specific script didn't work".